From cd818a9975d40f188b588ee30bac16bc23fd6311 Mon Sep 17 00:00:00 2001
From: Ming Liu <ming.liu@toradex.com>
Date: Mon, 14 Sep 2020 11:01:08 +0200
Subject: [PATCH] colibri_imx7: use distro boot by default

Use distro boot for eMMC as well as raw NAND modules by default. This
is required for TorizonCore.

Also set bootubivol to rootfs.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
Signed-off-by: Ming Liu <ming.liu@toradex.com>
---
 include/configs/colibri_imx7.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/include/configs/colibri_imx7.h b/include/configs/colibri_imx7.h
index 1b50dac83c..564f7b0371 100644
--- a/include/configs/colibri_imx7.h
+++ b/include/configs/colibri_imx7.h
@@ -108,8 +108,7 @@
 		"run fdt_fixup && bootz ${kernel_addr_r} - ${fdt_addr_r}\0" \
 
 #if defined(CONFIG_TARGET_COLIBRI_IMX7_NAND)
-#define CONFIG_BOOTCOMMAND "run ubiboot ; echo ; echo ubiboot failed ; " \
-	"run distro_bootcmd;"
+#define CONFIG_BOOTCOMMAND "run distro_bootcmd;"
 #define MODULE_EXTRA_ENV_SETTINGS \
 	"mtdparts=" CONFIG_MTDPARTS_DEFAULT "\0" \
 	UBI_BOOTCMD
@@ -121,6 +120,7 @@
 
 #if defined(CONFIG_TARGET_COLIBRI_IMX7_NAND)
 #define BOOT_TARGET_DEVICES(func) \
+	func(UBIFS, ubifs, 0) \
 	func(MMC, mmc, 0) \
 	func(USB, usb, 0) \
 	func(DHCP, dhcp, na)
@@ -142,6 +142,7 @@
 	"boot_file=zImage\0" \
 	"bootm_boot_mode=sec\0" \
 	"bootubipart=ubi\0" \
+	"bootubivol=rootfs\0" \
 	"console=ttymxc0\0" \
 	"defargs=\0" \
 	"fdt_board=eval-v3\0" \
-- 
2.29.0

